src/app/games/lasca/LascaMove.ts

```diff
@@ -0,0 +1,143 @@
+                if (jumpedOverCoords.contains(jumpedOverCoord)) {
+                    return MGPFallible.failure(LascaMoveFailure.CANNOT_CAPTURE_TWICE_THE_SAME_COORD());
+                }
+                jumpedOverCoords.add(jumpedOverCoord);
+            }
+            lastCoordOpt = MGPOptional.of(coord);
+        }
+        return MGPFallible.success(jumpedOverCoords);
+    }
+    public static fromStep(start: Coord, end: Coord): MGPFallible<LascaMove> {
+        if (LascaMove.isNotOnBoard(start)) {
+            return MGPFallible.failure(CoordFailure.OUT_OF_RANGE(start));
+        }
+        if (LascaMove.isNotOnBoard(end)) {
+            return MGPFallible.failure(CoordFailure.OUT_OF_RANGE(end));
+        }
+        const vector: Coord = start.getVectorToward(end);
+        if (Math.abs(vector.x) !== 1 || Math.abs(vector.y) !== 1) {
+            return MGPFallible.failure(LascaMoveFailure.MOVE_STEPS_MUST_BE_SINGLE_DIAGONAL());
+        }
+        return MGPFallible.success(new LascaMove([start, end], true));
+    }
+    public static encoder: MoveEncoder<LascaMove> = new class extends MoveEncoder<LascaMove> {
+        public encodeMove(move: LascaMove): JSONValueWithoutArray {
+            return {
+                coords: move.coords.map((coord: Coord): JSONValueWithoutArray => {
+                    return Coord.encoder.encode(coord) as JSONValueWithoutArray;
+                }),
+                isStep: move.isStep,
+            };
+        }
+        public decodeMove(encoded: JSONValueWithoutArray): LascaMove {
+            const casted: JSONObject = encoded as JSONObject;
+            assert(casted.coords != null, 'Encoded LascaMove should have a coords field');
+            assert(casted.isStep != null, 'Encoded LascaMove should have a isStep field');
+            const encodedCoords: JSONValueWithoutArray[] =
+                Utils.getNonNullable(casted.coords) as JSONValueWithoutArray[];
+            const coords: Coord[] = encodedCoords.map((x: JSONValue) => Coord.encoder.decode(x));
+            return new LascaMove(coords, casted.isStep as boolean);
+        }
+    };
+    public readonly coords: MGPSet<Coord>;
+
+    private constructor(coords: Coord[], public readonly isStep: boolean) {
+        super();
+        this.coords = new MGPSet(coords);
+    }
+    public toString(): string {
+        const coordStrings: string[] = this.coords.map((coord: Coord) => coord.toString());
+        const coordString: string = coordStrings.join(', ');
+        return 'LascaMove(' + coordString + ')';
+    }
+    public getRelation(other: LascaMove): 'EQUALITY' | 'PREFIX' | 'INEQUALITY' {
+        const thisLength: number = this.coords.size();
+        const otherLength: number = other.coords.size();
+        const minimalLength: number = Math.min(thisLength, otherLength);
+        for (let i: number = 0; i < minimalLength; i++) {
+            if (this.coords.getByIndex(i).equals(other.coords.getByIndex(i)) === false) return 'INEQUALITY';
```

Comment:
ça n'a pas de sens qu'un set soit un préfixe d'un autre d'abord, ensuite {1, 2} est égal à {2, 1}, mais ici ça dirait inégal !

Reply:
Oui, ça devrait être une liste sans doublon, je tickette la création de cette Collection et le refactor de ceci ?

Reply:
ou fais le dans ce ticket plutôt que d'ajouter des méthodes illogiques !